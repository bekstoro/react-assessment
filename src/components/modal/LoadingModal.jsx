import React from 'react';
import PropTypes from 'prop-types';

import Portal from '@components/portal/Portal';
import { LoadingIcon, LoadingText, SmallModalCentered } from '@components/styled-components/Modal';

export default function LoadingModal({ isOpen, closeModal }) {
  if (!isOpen) return null;

  return (
    <Portal closeModal={closeModal}>
      <SmallModalCentered>
        <LoadingIcon />
        <LoadingText>Data is loading</LoadingText>
      </SmallModalCentered>
    </Portal>
  );
}

LoadingModal.propTypes = {
  isOpen: PropTypes.bool.isRequired,
  closeModal: PropTypes.func.isRequired
};
