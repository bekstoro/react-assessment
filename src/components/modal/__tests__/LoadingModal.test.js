import React from 'react';
import { render, within } from '@testing-library/react';

import LoadingModal from '@components/modal/LoadingModal';

describe('LoadingModal', () => {
  beforeEach(() => {
    const modalRoot = document.createElement('div');
    modalRoot.setAttribute('id', 'app');
    document.body.appendChild(modalRoot);
  });

  it('LoadingModal renders correctly', () => {
    render(<LoadingModal isOpen closeModal={jest.fn()} />);
    const { getByTestId } = within(document.getElementById('app'));
    expect(getByTestId('loading-modal')).toMatchSnapshot();
  });

  it('LoadingModal should return empty div', () => {
    const { container } = render(<LoadingModal isOpen={false} closeModal={jest.fn()} />);
    expect(container).toMatchSnapshot();
  });
});
