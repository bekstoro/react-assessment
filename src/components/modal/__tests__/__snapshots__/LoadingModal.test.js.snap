// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingModal LoadingModal renders correctly 1`] = `
<div
  class="Modal__BaseModal-sc-1081dim-0 Modal__SmallModal-sc-1081dim-2 Modal__SmallModalCentered-sc-1081dim-3 cyuJJj bOgRmL iwQtpG"
  data-testid="loading-modal"
>
  <div
    class="Modal__LoadingIcon-sc-1081dim-4 cmqSzZ"
  />
  <h3
    class="Modal__LoadingText-sc-1081dim-5 gTCjkZ"
  >
    Data is loading
  </h3>
</div>
`;

exports[`LoadingModal LoadingModal should return empty div 1`] = `<div />`;
